cookietemple/create/template_creator.py

```diff
@@ -293,22 +283,43 @@ def create_common_files(self) -> None:
         # change to recent cwd so lint etc can run properly
         os.chdir(str(cwd_project))
 
-    def readthedocs_slug_already_exists(self, project_name: str) -> bool:
+    def check_pypi_readthedocs(self, host, dot_cookietemple) -> None:
+        """
+        Main function that calls the checks for PyPi and readthedocs
+        """
+        while TemplateCreator.call_host(host, self.creator_ctx.project_name) and not dot_cookietemple:
+            print(f'[bold red]A project named {self.creator_ctx.project_name} already exists at {host}!')
+            if cookietemple_questionary_or_dot_cookietemple(function='confirm',
+                                                            question='Do you want to choose another name for your project?\n'
+                                                                     f'Otherwise you will not be able to host your project at {host}!', default='Yes'):
+                self.creator_ctx.project_name = cookietemple_questionary_or_dot_cookietemple(function='text',
+                                                                                             question='Project name',
+                                                                                             default='Exploding Springfield')
+            # break if the project should be named anyways
+            else:
+                break
+
+    @staticmethod
+    def call_host(host: str, project_name: str) -> bool:
         """
-        Test whether there´s already a project with the same name on readthedocs
+        Make a GET request to the host to check whether a project with this name already exists.
+        :param host The host (either PyPi or readthedocs)
         :param project_name Name of the project the user wants to create
+
+        :return: Whether request was successful (name already taken on host) or not
         """
-        log.debug(f'Look up {project_name} at readthedocs.io .')
-        print(f'[bold blue]Looking up {project_name} at readthedocs.io!')
+        print(f'[bold blue]Looking up {project_name} at {host}!')
+        url = 'https://' + (f'pypi.org/project/{project_name.replace(" ", "")}' if host == "PyPi" else f'{project_name.replace(" ", "")}.readthedocs.io')
```

Comment:
you generalized this function, but now only allow pypi and readthedocs

the name of the function does not give that away

-> if it s no pypi or readthedocs you should catch that earlier and inform the user

Reply:
```
-> if it s no pypi or readthedocs you should catch that earlier and inform the user```

You mean in terms of scalability? Because this function is only called with either PyPi or readthedocs